lib/SIL/Utils/SILCloner.cpp

```diff
@@ -0,0 +1,107 @@
+//===--- SILCloner.h - Defines the SILCloner class --------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the SILCloner class, used for cloning SIL instructions.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/SIL/SILCloner.h"
+#include "swift/SIL/SILFunctionBuilder.h"
+
+namespace swift {
+
+std::unique_ptr<SILModule> cloneModule(SILModule &originalModule) {
+  // This implementation is based on llvm::CloneModule which can be found here:
+  // https://llvm.org/doxygen/namespacellvm.html#ab371d6b308eb9772bdec63cf7a041407
+
+  assert((originalModule.getStage() != SILStage::Canonical ||
+          originalModule.getStage() != SILStage::Lowered) &&
+         "cloneModule doesn't support SILStage::Lowered.");
```

Comment:
It looks like this assertion condition is always going to be true, and the assertion string is out-of-sync from the condition. Could you clarify what is intended to be checked here?